src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -21,11 +25,14 @@ public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
     private final BookSpecificationBuilder bookSpecificationBuilder;
+    private final CategoryRepository categoryRepository;
 
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
-        return bookMapper.toDto(
-                bookRepository.save(bookMapper.toModel(requestDto)));
+        Set<Category> categories = categoryRepository.findByIdIn(requestDto.getCategoryIds());
+        Book book = bookMapper.toModel(requestDto);
+        book.setCategories(categories);
```

Comment:
```suggestion
        Book book = bookMapper.toModel(requestDto);
```
add to mapper aftermapping method. We do not need to fetch categories from DB - just create objects using Ids